preprocess.py

```diff
@@ -90,49 +48,45 @@ def dict_path(lang):
         return dest_path("dict", lang) + ".txt"
 
```

Comment:
Let's add a local helper function to build the dictionary, which can include all the extra options (currently missing below), such as threshold, nwords, padding_factor, etc.
```
def build_dictionary(filenames, src=False, tgt=False):
    assert src ^ tgt
    return task.build_dictionary(
        filenames,
        workers=args.workers,
        threshold=args.thresholdsrc if src else args.thresholdtgt,
        nwords=args.nwordssrc if src else args.nwordstgt,
        padding_factor=args.padding_factor,
    )
```